include/drivers/sensor.h

```diff
@@ -161,6 +161,8 @@ enum sensor_channel {
 	SENSOR_CHAN_GAUGE_AVG_POWER,
 	/** State of health measurement in % **/
 	SENSOR_CHAN_GAUGE_STATE_OF_HEALTH,
+	/** Weight. Units depend on SENSOR_ATTR_SLOPE value. */
+	SENSOR_CHAN_WEIGHT,
```

Comment:
I'd like to see a standard unit here, not variable

Reply:
I would really like to be able to force grams as per SI but I believe that it is not possible.
You see the HX711 is simply an ADC, what I get from the sensor is an 24 bit value.
The sensor it self is agnostic to the units, unlike gyroscopes or humidity sensors.

The actual raw sensor value depends on the mechanical and material aspects of the weight scale and also the specific load cells used.

To get an actual measurement which makes any sense in any unit the following formula is used :

**_weight = slope * (raw - offset)_**

**offset** is an raw measurement used for zeroing obtained by **hx711_tare**.
**slope**, well it is the slope of the linear change and is obtained by **hx711_calibrate**.

**hx711_calibrate** requires a target parameter as an **u32_t**.

After zeroing, the user invokes this function and provides a known weight in the unit of his choosing.

If the **_target_** parameter represents _grams_, then **val->val1** from **hx711_channel_get** represents grams.
If it represents _pounds_, then **val->val1** represents pounds.

If I could force **u32_t** to represent **grams** then is could force **SENSOR_CHAN_WEIGHT** to be **grams**.
But as it is, I really cannot force the user to represent grams for the target parameter.

Of course I could **typedef u32_t grams **, but this still would not restrict people from using imperial, tons, Kg or anything they would like.

(**offset** and **tare** can also be defined through **Kconfig** in the case of a controlled manufacturing process of the weight scale.)

Reply:
I'm not sure what alternatives to suggest here, but a variable unit just doesn't make sense with the sensor API. Could you use device tree properties, and set them at the board or application level?

@microbuilder @pabigot any ideas?

Reply:
@MaureenHelm I agree that we should have specific SI units and scale associated with channels, grams making the most sense here.

Perhaps we have to break this SI unit expectation for certain edge cases, and clearly document that in the sensor driver, but couldn't we use a device tree property to define a `whatever` to `g` coefficient with this sensor, that the firmware author should know for their own hardware, and we can apply that to return `g`?

We really need to come back to the sensor API sometime in the near future, though. I had similar issues designing a spectrometer recently, including a Zephyr driver for the spectral sensor, and there is no good way to handle that data today, similar to this example.

Some units in the sensor API today are just outright wrong as well, such as for light:

```
/** Illuminance in infra-red spectrum, in lux. */
SENSOR_CHAN_IR,
/** Illuminance in red spectrum, in lux. */
SENSOR_CHAN_RED,
/** Illuminance in green spectrum, in lux. */
SENSOR_CHAN_GREEN,
/** Illuminance in blue spectrum, in lux. */
SENSOR_CHAN_BLUE,
```

Lux, as a photometric value, is only relevant to human vision and visible light, and is based on a model that estimates how the human eye perceives brightness with the red, green and blue cones in our eyes. As such, lux has no meaning for individual color channels, and certainly not for IR which it outside the human visible range, and these should all be radiometric units like irradiance.

Reply:
From what I can see from the datasheet and SparkFun tutorials it should work to specify that the unit for `SENSOR_CHAN_WEIGHT` is grams.  Then `avia_hx711_calibrate()` should be given the calibration weight as a `struct sensor_value` in grams.  This can be documented in the (to be provided) documentation for these functions.

A user may choose to provide a calibration value in an arbitrary unit and everything will work, with the understanding that doing so is not compliant with Zephyr's sensor framework.

Reply:
There seems no way to force the API to work with **only** grams. 
I agree with @pabigot  that documenting that the values passed to calibrate and the value returned from **sensor_channel_get()** are in grams.

Documentation would then be the only **_line of defence_** against someone using another unit.

I would be a shame though to not let people know they can use **any unit** they like right away, and make them write conversion functions :)

Currently I document that the unit returned is the same as the unit provided during calibration:

```
/**
 * @brief Callibrate the HX711.
 *
 * Given a target value of aknown weight the slope gets calculated.
 * This is actually unit agnostic.
 * If the target weight is given in grams, lb, Kg or any other weight unit,
 * the slope will be calculated accordingly.
 *
 * @param dev Pointer to the hx711 device structure
 * @param target Target weight in whatever unit.
 * @param readings Number of readings to take for calibration.
 *        5~10 readings should be enough, although more are allowed.
 * @retval The slope value
 *
 */
```

Reply:
I changed documentation indicating that WEIGHT is in grams.
The possibility to use other units is described.
```
/**
 * @brief Callibrate the HX711.
 *
 * Given a target value of aknown weight the slope gets calculated.
 * This is actually unit agnostic.
 * If the target weight is given in grams, lb, Kg or any other weight unit,
 * the slope will be calculated accordingly.
 *
 * @param dev Pointer to the hx711 device structure
 * @param target Target weight in grams.
 *        If target is represented in another unit (lb, oz, Kg) then the
 *        value returned by sensor_channel_get() will represent that unit.
 * @param readings Number of readings to take for calibration.
 *        5~10 readings should be enough, although more are allowed.
 * @retval The slope value
 *
 */
```

I believe that with this change the solution can be accepted.